### PR TITLE
[core] Remove usages of deprecated PMD.VERSION

### PR DESCRIPTION
**Pull request by jsotuyod**
- Take the chance to add some helpers to PMDVersion that seem needed
 - Have CodeClimate renderer point to pmd.github.io rather than SF

### CODE REVIEW

File: pmd-core/src/main/java/net/sourceforge/pmd/renderers/CodeClimateRenderer.java [hunk 1]
```diff
@@ -45,9 +46,9 @@ public CodeClimateRenderer() {
     }
 
     private static String getPmdDeveloperURL() {
-        String url = "http://pmd.github.io/pmd-" + PMD.VERSION + "/customizing/pmd-developer.html";
-        if (PMD.VERSION.contains("SNAPSHOT") || "unknown".equals(PMD.VERSION)) {
-            url = "http://pmd.sourceforge.net/snapshot/customizing/pmd-developer.html";
+        String url = "https://pmd.github.io/pmd-" + PMDVersion.VERSION + "/customizing/pmd-developer.html";
+        if (PMDVersion.isSnapshot() || PMDVersion.isUnknown()) {
+            url = "https://pmd.github.io/latest/customizing/pmd-developer.html";
```
Comment (oowekyala) [inline pmd-core/src/main/java/net/sourceforge/pmd/renderers/CodeClimateRenderer.java]: This URL is a 404. Would it be `https://pmd.github.io/latest/pmd_devdocs_development.html` ?

Comment (jsotuyod) [reply pmd-core/src/main/java/net/sourceforge/pmd/renderers/CodeClimateRenderer.java]: Good catch! Thanks!

**Comment by jsotuyod** [comment]
@oowekyala done!

File: pmd-core/src/main/java/net/sourceforge/pmd/renderers/CodeClimateRenderer.java [hunk 2]
```diff
@@ -36,18 +37,17 @@
...
     }
 
-    private static String getPmdDeveloperURL() {
-        String url = "http://pmd.github.io/pmd-" + PMD.VERSION + "/customizing/pmd-developer.html";
-        if (PMD.VERSION.contains("SNAPSHOT") || "unknown".equals(PMD.VERSION)) {
-            url = "http://pmd.sourceforge.net/snapshot/customizing/pmd-developer.html";
+    private static String getPmdPropertiesURL() {
+        String url = "https://pmd.github.io/pmd-" + PMDVersion.VERSION + "/pmd_devdocs_working_with_properties.html";
+        if (PMDVersion.isSnapshot() || PMDVersion.isUnknown()) {
+            url = "https://pmd.github.io/latest/pmd_devdocs_working_with_properties.html";
```
Comment (oowekyala) [inline pmd-core/src/main/java/net/sourceforge/pmd/renderers/CodeClimateRenderer.java]: We should set up a reminder to update that when the new version of the site goes online.

**Comment by oowekyala** [comment]
@jsotuyod Quick question, should I add an entry in the release notes under Fixed or External contributions ? This PR has no corresponding issue

**Comment by jsotuyod** [comment]
@oowekyala no, we don't put anything under external contributors for our own PRs. As for the rest of the release notes, we don't mention it if:

1. it's not closing an issue
2. it's not changing a user-facing behavior

in this case, there is nothing worth adding to the changelog.

**Comment by oowekyala** [comment]
@jsotuyod Ok thanks. That may be worth adding to the Merging PRs docs for future reference

**Pull request merged**